Source/Dafny/Dafny.atg

```diff
@@ -4322,13 +4359,17 @@ ActualBindings<.List<ActualBinding> bindings.>
   .
 
 ActualBinding<out ActualBinding binding>
-= (. IToken id = null; Expression e; .)
+= (. IToken id = null; Expression e; bool isGhost = false; .)
+  [
+    // TODO(fpoli): The linear-types branch uses `IF(la.kind == _ghost && !IsMatch() && !IsVar())` here. Any clue why they needed that?
+    "ghost" (. isGhost = true; .)
+  ]
   [ IF(IsBinding())
     NoUSIdentOrDigits<out id>
     ":="
   ]
   Expression<out e, true, true>
-  (. binding = new ActualBinding(id, e); .)
+  (. binding = new ActualBinding(id, e, isGhost); .)
```

Comment:
This is good. The `ghost` keyword should be allowed only if the `ActualBinding` denotes a component of a tuple. The Resolver should give an error if `ghost` is used for any other purpose. (Alternatively, you can add a parameter `bool allowGhostKeyword` to the `ActualBinding` grammar production and check the condition it here in the parser.)

Reply:
I moved the `[ "ghost" ]` to `TupleArgs`. `ActualBinding` now has `isGhost` as a parameter.